spring-cloud-gcp-data-spanner/src/main/java/org/springframework/cloud/gcp/data/spanner/repository/query/SpannerStatementQueryExecutor.java

```diff
@@ -193,7 +214,13 @@ private static StringBuilder buildSelect(
 		if (tree.isDistinct()) {
 			stringBuilder.append("DISTINCT ");
 		}
-		stringBuilder.append(getColumnsStringForSelect(spannerPersistentEntity) + " ");
+		if (tree.isExistsProjection() || tree.isCountProjection()) {
+			stringBuilder.append("COUNT(1) ");
```

Comment:
Should we do COUNT(DISTINCT *) instead. Please check other Spring Data implementations.